### PR TITLE
fix(multipart processor): capture original file name

### PR DESCRIPTION
**Pull request by bxlxx**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [x] My code includes positive and negative tests.
- [x] I have an appropriate description with correct grammar.
- [x] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

**Review by jptosso — CHANGES_REQUESTED** [review]
Hey! Please add some tests on why p.FileName was changed to p.OriginFileName. But it does total sense.

About cloning the multipart/multipart.go file from the golang project, is there any major reason? 

There is a hack we can use to access private functions:

```go
    _ "unsafe"
)

//go:linkname contains strconv.contains
func contains(s string, c byte) bool
```

@fzipi what do you think?

**Review by fzipi — CHANGES_REQUESTED** [review]
I don't see any justification on copying the upstream multipart.

While I see the reasons we might do this, it should be documented really well. Why? What we need to change on that file to solve which problem?

**Comment by fzipi** [comment]
@bxlxx If you add the `Fixes #226` while creating the PR everything gets linked and it is clear what you are trying to solve. Also, in this case, a comment also should have helped to understand why you are including the multipart from upstream, so when people read the code afterward don't have these doubts and questions.

**Comment by jcchavezs** [comment]
I'd really avoid using of reflection for a function that is simple and hasn't change in a while. I opened a proposal in this PR https://github.com/corazawaf/coraza/pull/229/files including a test.

**Comment by bxlxx** [comment]
My personal idea is that I only use ```reflect.TypeOf``` without modifying data through ```reflect```, which will not have much impact on performance. And, ```p.FormName``` will be used in the subsequent process. It will call ```p.parseContentDisposition()``` again to assign value to ```p.dispositionParams``` field. I think it's a little redundant.

File: seclang/rules_test.go [hunk 1]
```diff
@@ -535,3 +536,66 @@ func TestRxCapture(t *testing.T) {
...
+	if err != nil {
+		t.Error()
+		return
+	}
+	it, err := tx.ProcessRequestBody()
+	if err != nil {
+		t.Error()
+		return
+	}
+	if it == nil {
```
Comment (jcchavezs) [inline seclang/rules_test.go:598]: What exactly do we expect in this interruption?

Comment (bxlxx) [reply seclang/rules_test.go:598]: Just used to verify that the rule was triggered.

**Comment by jptosso** [comment]
Replaced with #229

**Pull request closed without merging**